confidant/routes/v1.py

```diff
@@ -366,6 +370,7 @@ def get_credential_list():
 
 @app.route('/v1/credentials/<id>', methods=['GET'])
 @authnz.require_auth
+@authnz.require_role(role='read_only')
```

Comment:
Since you need to check against a resource, it may be better to have this as a function call inside of the function, rather than as a decorator, as you'll be able to pass-in the credential model instance, rather than doing a double fetch.

Reply:
Also, if you do it as a function call, you could make it generic and only pass in metadata, which would make the function work with anything that has metadata (credentials, services, blind credentials, etc).

Reply:
do you mean a function call within the `get_credential()`  and `update_credential()` functions ? 

example:
```
authnz.require_role(Credential.metadata)
```

Reply:
yep!